Artifact "resonance" penalties from carrying too many (#61341)
CleverRaven/Cataclysm-DDA: doc/ARTIFACTS.md

## Resonance
Resonance is an enchantment assigned to artifacts, and is always equal to the artifact's power, being assigned according to the final power of the artifact after its main properties have been generated.  Although artifacts do not have to have resonance (and will not have resonance unless explicitly assigned) all vanilla artifacts with the sole exception of the "Exposed-Wiring Prototype" have resonance, as should any future artifacts added to the game.  Resonance is always at least zero, so a purely detrimental artifact cannot be used to "lower" your resonance although a mixed artifact is less resonant than a purely positive one.

Resonance is assigned to artifacts in the JSON responsible for adding them to mapgen, where artifacts are given a power level and a set of procgen data to draw from. If flagged as true, this artifact will have resonance. If it does not include this optional flag, it will default to false.

> Resonance represents the potential dangers of artifacts on the human body to counterbalance their positive effects.  For the purposes of lore, artifacts should primarily cause such resonance effects when they are affecting a human body - when held close at hand, similar to artifacts in S.T.A.L.K.E.R.  Resonance effects are currently hardcoded, and scale with the amount of resonance you have, beginning at 2,000 resonance (2,000 total power of artifacts carried).  Resonance of 2,000 causes an effect roughly every hour, but effect also occur more frequently the more resonance you have, becoming 100% more common for every whole 2,000 resonance. There are 4 "tiers" of effects in increasing severity kicking in at 2,000, 4,500, 7,500, 12,500 resonance. Each "tier" of resonance has a 50% chance to downgrade to a lower tier of severity, recursively, so lower-severity effects never go away.

Reviewer: Don't use references to other games to explain effects.   I've never played STALKER and might not ever.
```suggestion
Resonance represents the potential dangers of artifacts on the human body to counterbalance their positive effects.  For the purposes of lore, artifacts should primarily cause such resonance effects when they are affecting a human body - such as when carried or wielded by the PC.  Resonance effects are currently hardcoded, and scale with the amount of resonance you have, beginning at 2,000 resonance (2,000 total power of artifacts carried).  Resonance of 2,000 causes an effect roughly every hour, but effect also occur more frequently the more resonance you have, becoming 100% more common for every whole 2,000 resonance. There are 4 "tiers" of effects in increasing severity kicking in at 2,000, 4,500, 7,500, 12,500 resonance. Each "tier" of resonance has a 50% chance to downgrade to a lower tier of severity, recursively, so lower-severity effects never go away.
```
Give an example please.   I'm not sure if you are saying that higher tier effects have a chance to add additional lower tier effects or that there is a chance for higher tier effects to transform into a lower tier effect instead of applying the higher tier effect.

> These effects should be relatively vague - it should not be explicitly clear that carrying multiple artifacts is what's causing a player's misfortune, but it should be implied due to the effects not happening until the player picks up a bunch of cool-looking napkins and rubick's cubes.  Tier 1 effects are nausea, low-strength hallucinations, and minor pain. Tier 2 effects are minor radiation (blocked by rad resistant gear), safe micro-teleports and a very loud noise.  Tier 3 effects include severe pain, severe radiation (not fully blockable by gear), and unsafe teleportation.  Tier 4 effects include hounds of Tindalos, incorporeal, and creating a tear in reality. The resonance factor of an artifact should be measurable, even if the other properties of the artifact are not, although at time of writing there is no way to do so.

Reviewer: Do these resonance effects also occur to NPCs? I'm not even sure Artifacts affect NPCs in general, they probably do but I've never checked.  What happens if you give Smokes 40k worth of artifact power?

Author: Smokes will asplode, yes. Suffer.cpp occurs on NPCs and they are also affected by enchants.
